docs/streaming-functions.md

```diff
@@ -0,0 +1,146 @@
+# Data Stream events
+
+Kubeless lets you trigger any Kubeless function in response to ingested records into a data stream. Kubeless currently supports AWS Kinesis streaming service. 
+
+## AWS Kinesis
+
+To trigger Kubeless functions in response to ingested records into the AWS kinesis stream you need to deploy Kubeless AWS Kinesis trigger controlle. Please use this manifest to deploy Kubeless AWS Kinesis triggers controller.
+
+```
+kubectl create -f https://github.com/kubeless/kubeless/releases/download/$RELEASE/kinesis-$RELEASE.yaml
+```
+
+Once you deploy the manifest you shall see Kinesis trigger controller running in the Kubeless namespace as below.
+
+```console
+± kubectl get pods -n kubeless
+NAME                                           READY     STATUS    RESTARTS   AGE
+kinesis-trigger-controller-65c78f9f44-v5flq    1/1       Running   0          1h
+kubeless-controller-manager-6b7cdcdc76-x6gsd   1/1       Running   0          13h
+```
+
+You shall also notice a CRD resource type `kinesistriggers.kubeless.io` created as below.
+```console
+± kubectl get crd
+NAME                          AGE
+cronjobtriggers.kubeless.io   13h
+functions.kubeless.io         13h
+httptriggers.kubeless.io      13h
+kinesistriggers.kubeless.io   13h
+```
+
+Kubeless cli lets you create Kubeless triggers of Kinesis type. Kubeless cli provides necessary functionality to manage life cycle of Kinesis triggers.
+
+```console
+± kubeless trigger kinesis
+kinesis trigger command allows user to create, list, update, delete Kinesis triggers running on Kubeless
+
+Usage:
+  kubeless trigger kinesis SUBCOMMAND [flags]
+  kubeless trigger kinesis [command]
+
+Available Commands:
+  create      Create a Kinesis trigger
+  delete      Delete a Kinesis trigger
+  list        list all Kinesis triggers deployed to Kubeless
+  publish     publish message to a Kinesis stream
+  update      Update a Kinesis trigger
+
+Flags:
+  -h, --help   help for kinesis
+
+Use "kubeless trigger kinesis [command] --help" for more information about a command.
+```
+
+In order to deploy a Kinesis trigger and associate a Kubeless function to be invoked in response to ingested records in Kinesis data stream, you need to first let Kubeless know the credentials required to acess your AWS Kinesis stream. Kubeless will leverage Kubernetes secrets to store the credentails in the cluster and use them to acess the Kinesis stream.
+
+First you need to creat Kubernetes secret that can store you AWS `aws_access_key_id` and `aws_secret_access_key`. Usually if you are using AWS cli your keys will be present in `~/.aws/credentials` or you can create access keys from AWS console.
+
+```console
+kubectl create secret generic ec2 --from-literal=aws_access_key_id=ABCAIFDB7GEGGD37HN5Q --from-literal=aws_secret_access_key=gAckl2wtNrY9aaR/OvzAOCaAg6wSkABCsyXazXrx
```

Comment:
I would put a placeholder here for the id and key just to avoid people trying to use the same credentials :)